src/get.cpp

```diff
@@ -95,21 +97,22 @@ Rcpp::List C_time_get(const NumericVector& dt,
 
     }
 
-  List out = DataFrame::create();
-  CharacterVector names;
+  List lst(N_comps);
+  CharacterVector names(N_comps);
 
-  for (std::string comp: comps) {
-    if (comp == "year") { out.push_back(year); names.push_back("year"); continue; };
-    if (comp == "month") { out.push_back(month); names.push_back("month"); continue; };
-    if (comp == "yday") { out.push_back(yday); names.push_back("yday"); continue; };
-    if (comp == "day") { out.push_back(mday); names.push_back("day"); continue; };
-    if (comp == "mday") { out.push_back(mday); names.push_back("mday"); continue; };
-    if (comp == "wday") { out.push_back(wday); names.push_back("wday"); continue; };
-    if (comp == "hour") { out.push_back(hour); names.push_back("hour"); continue; };
-    if (comp == "minute") { out.push_back(minute); names.push_back("minute"); continue; };
-    if (comp == "second") { out.push_back(second); names.push_back("second"); continue; };
-  }
+  R_len_t pos = 0;
+
+  if (do_year) {lst[pos] = year; names[pos] = "year"; pos++;};
+  if (do_month) {lst[pos] = month; names[pos] = "month"; pos++;};
+  if (do_yday) {lst[pos] = yday; names[pos] = "yday"; pos++;};
+  if (do_mday) {lst[pos] = mday; names[pos] = "mday"; pos++;};
+  if (do_wday) {lst[pos] = wday; names[pos] = "wday"; pos++;};
+  if (do_hour) {lst[pos] = hour; names[pos] = "hour"; pos++;};
+  if (do_minute) {lst[pos] = minute; names[pos] = "minute"; pos++;};
+  if (do_second) {lst[pos] = second; names[pos] = "second"; pos++;};
+
+  lst.attr("names") = names;
 
-  out.attr("names") = names;
+  DataFrame out = Rcpp::as<DataFrame>(lst);
```

Comment:
We can return an actual data frame like this, so we don't need `as.data.frame()` on the R side

Reply:
Indeed, that was the original intent but I forgot how to do it :) That conversion might be even automatic as the return type is DataFrame.